### PR TITLE
Feature/instance pages

### PR DESCRIPTION
**Pull request by borysyuk**

### CODE REVIEW

File: test/handlers/search.js [hunk 1]
```diff
@@ -81,6 +81,19 @@ describe('search handler', () => {
...
+        mockApiResponses();
+        return server.inject('/search?q=test&page=aa')
+          .then((_response) => {
+            _response.statusCode.should.equal(400)
+          });
+
+        return server.inject('/search')
+          .then((_response) => {
+            _response.statusCode.should.equal(502);
+          });
```
Comment (vitorbaptista) [inline test/handlers/search.js]: These lines will never be executed because there's a `return` before them.

File: routes/index.js [hunk 2]
```diff
@@ -35,6 +35,28 @@ const routes = [
     method: 'GET',
     handler: require('../handlers/search'),
   },
+
+  {
+    path: '/problems/{id?}',
```
Comment (vitorbaptista) [inline routes/index.js]: As the `id` is required, use `/problems/{id}` instead (same for the other routes)

File: views/interventions-details.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{% extends "layouts/two-columns.html" %}
+
+{% block bannerFooter %}
+<ul class="overview-list">
+  <li>
+      <span class="date">
+        Registration: {{ intervention.registration_date | formatDate }}
```
Comment (vitorbaptista) [inline views/interventions-details.html]: Where does this `registration_date` come from?

File: routes/index.js (hunk 2 above)
Comment (borysyuk) [inline routes/index.js]: i copied from this route path: '/trials/{id?}',
I think, id is required here too, right?

Comment (vitorbaptista) [inline routes/index.js]: Yes, that was the remnants of a `/trials` route. I fixed it there as well in another pull request :+1:

**Comment by vitorbaptista** [comment]
@borysyuk Could you fix the conflicts with master? Also, your IDE re-formatted the code again.

**Comment by borysyuk** [comment]
@vitorbaptista I have merged and resolved conflicts

**Comment by vitorbaptista** [comment]
Cool! I'll merge it as soon as https://github.com/opentrials/api/pull/36 is done, so we don't end up with an interface accessing an API that doesn't exist.

**Comment by vitorbaptista** [comment]
Merged in https://github.com/opentrials/opentrials/commit/48ad147b5dd2073e93dd5b7cab4496bd7c708f43 :+1:

**Pull request closed without merging**